Blog plugin update (#4906)
jenkins-infra/jenkins.io: content/blog/2022/02/2022-02-10-last-plugin-version-not-installable.adoc

---
layout: post
title: "Why can't I install the latest plugin on my Jenkins LTS?"
tags:
- jenkins
- plugin
- upgrade
- developer
authors:
- jmMeessen
description: >

>   This blogpost explains why the latest version of a plugin can't be installed on Jenkins in some circumstances.
> ---

Reviewer: An opengraph image is used when the blog post is shared on social media (like twitter) and increases the reading rate .

This reuses the image from inside the blogpost.  If you prefer a different image, I've attached an alternative to this comment (from the [Jenkins Social Media Covers slide deck](https://docs.google.com/presentation/d/1Q1PgNnRTgzBpVRXPqQo3PudzCa2eoc6_1_NRjFRMLrU/edit?usp=sharing)).

```suggestion
  This blogpost explains why the latest version of a plugin can't be installed on Jenkins in some circumstances.
opengraph:
  image: /images/post-images/2022-02-10-last-plugin-version-not-installable/plugin-not-installable.png
---
```

![update-visible-but-cannot-install](https://user-images.githubusercontent.com/156685/153264815-6507e1f9-9715-40af-9af2-ed781e8360c5.png)

The message indicates to Jenkins Administrators that a newer plugin version is available 
but that the installed Jenkins Core doesn't provide the necessary pre-requisite.

**Why is this happening now?**
Did I miss something?

As part of the continuous improvement of Jenkins, some critical and high impact changes are introduced.
This is typically the case of GUI changes (icons, fonts, layout) as is currently happening.
It is also the case for new core features that can be consumed by plugins.

Such changes are typically first introduced in the cutting edge Jenkins "weekly release".
Plugins are adapted to take advantage of these changes.
The minimal Jenkins release requirement for those new plugin version is set to the "cutting-edge" release to prevent incompatibility.
This causes a clear requirement disruption compared to the generally slow and shallow plugin requirement evolution.

It will require some field-testing time before these major changes will make it into the LTS Jenkins version.
In the mean time, the new plugin will have to wait before it can be installed on that release line.

> If you _need_ to install that latest plugin version because it fixes also that annoying bug, you will have to decide whether it is worth taking the risk to install the less field-tested weekly release.

Reviewer 2: Please no trash-talking the weekly line. Those aren't "nightly builds", but actual versioned and supported releases.

Author: nuanced the statement, following @daniel-beck remark.

> If you _need_ to install that latest plugin version because it fixes also that annoying bug, you will have to decide whether it is worth taking the risk to install the less field-tested weekly release.

Reviewer: ```suggestion
If you _need_ to install that latest plugin version because it fixes also that annoying bug, you will have to install the weekly release.
```

Author: @MarkEWaite can you have a look to the reformulation I proposed, based on Daniel's suggestion: `If you _need_ to install that latest plugin version because it fixes also that annoying bug, you will have to evaluate whether it is worth switching to the weekly release line.`

Reviewer: That looks great to me.

---
layout: post
title: "Why can't I install the latest plugin on my Jenkins LTS?"
tags:
- jenkins
- plugin
- upgrade
- developer
authors:
- jmMeessen
description: >
  This blogpost explains why the latest version of a plugin can't be installed on Jenkins in some circumstances.
opengraph:
  image: /images/post-images/2022-02-10-last-plugin-version-not-installable/summary_picture.png
---

I regularly go, as all Jenkins admin do, to my system's Plugin Management page to see what plugins can be upgraded.

> Although I am running the latest link:/download/lts/[LTS version], **I can't update some key plugins** (example the Junit plugin).

Reviewer 2: ```suggestion
Although I am running the latest link:/download/lts/[LTS version], **I can't update some key plugins** (for example the JUnit plugin).
```

> This is even more intriguing as some are part of the recommended plugins (default plugin choice).

Reviewer 2: ```suggestion
This is even more intriguing as some are part of the plugins installed by default by the setup wizard.
```

No "install" tick box and the plugin is labeled "unavailable".
There is a note saying "_This version of the plugin exists but it is not being offered for installation, so the latest bug fixes or features are not available to you. This is typically the case when plugin requirements, e.g. a recent version of Jenkins, are not satisfied. If you are using the latest version of Jenkins offered to you, newer plugin releases may not be available to your release line yet. See the plugin documentation for information about its requirements._".

image:/images/post-images/2022-02-10-last-plugin-version-not-installable/plugin-not-installable.png[pluginView]

> The message indicates to Jenkins Administrators that a newer plugin version is available 

Reviewer 2: ```suggestion
The message indicates to Jenkins administrators that a newer plugin version is available 
```

> but that the installed Jenkins core doesn't provide the necessary prerequisite.

Reviewer 2: ```suggestion
but that the installed Jenkins core doesn't provide a necessary prerequisite – typically being too old.
```

**Why is this happening now?**
Did I miss something?

As part of the continuous improvement of Jenkins, some critical and high impact changes are introduced.
This is typically the case of GUI changes (icons, fonts, layout) as is currently happening.
It is also the case for new core features that can be consumed by plugins.

> Such changes are typically first introduced in the cutting edge Jenkins "weekly release".

Reviewer 2: Unusual to put it in quotes, I don't think we do that anywhere else. It's just the name.

Plugins are adapted to take advantage of these changes.
The minimal Jenkins version requirement for those new plugin releases is set to the "cutting-edge" release to prevent incompatibility.

> This causes a clear requirement disruption compared to the generally slow and shallow plugin requirement evolution.

Reviewer 2: Why shallow?

Author: in the sense "not deep"/"fundamental"
Will look for a better word.

> It will require some field-testing time before these major changes will make it into the LTS Jenkins version.

Reviewer 2: This could be misinterpreted to claim we backport large-scale changes when we don't.

Author: This was my perception of the relationship between LTS and Weekly. Will need to further polish this.

In the meantime, the new plugin will have to wait before it can be installed on that release line.

> If you _need_ to install that latest plugin version because it also fixes that annoying bug, you will have to evaluate whether it is worth switching to the weekly release line.

Reviewer 2: Could link to https://www.jenkins.io/download/lts/#switching-from-lts-to-weekly
